### PR TITLE
[codesign] Add pinned xcode version as property to mac android aot engine

### PR DESCRIPTION
**Pull request by XilaiZhang**
context: https://github.com/flutter/flutter/issues/125570#issuecomment-1523823942

Pin xcode version for the six sub builds in mac android aot engine

### CODE REVIEW

File: ci/builders/mac_android_aot_engine.json [hunk 1]
```diff
@@ -32,7 +32,13 @@
...
                     "flutter/shell/platform/android:gen_snapshot"
                 ]
-            }
+            },
+            "properties": {
+                "$flutter/osx_sdk": {
+                    "sdk_version": "14c18"
+                }
+            },
+            "tests": []
```
Comment (godofredoc) [inline ci/builders/mac_android_aot_engine.json:41]: rm

File: ci/builders/mac_android_aot_engine.json [hunk 2]
```diff
@@ -67,7 +73,13 @@
...
```
Comment (godofredoc) [inline ci/builders/mac_android_aot_engine.json:82]: rm

**Review by godofredoc — APPROVED** [review]

**Comment by XilaiZhang** [comment]
As a side note, I was testing it with `led get-build 8782752472421441633 | led edit -pa git_ref='refs/pull/41522/head' | led edit -pa git_url='https://github.com/flutter/engine' | led edit-system -p 20 | led launch` but looks like [the led run](https://ci.chromium.org/raw/build/logs.chromium.org/flutter/led/xilaizhang_google.com/8f9176a0cfa35574242111a77fe1212429ffd5ad8d6071361611f466f263b619/+/build.proto?server=chromium-swarm.appspot.com) did not identify [5a0b898](https://github.com/flutter/engine/pull/41522/commits/5a0b898c3addaff97d3778cf8e097f5cff2bd71a) as on a release branch based on our logic of [current_commit_branches](https://flutter.googlesource.com/recipes/+/refs/heads/main/recipe_modules/repo_util/api.py#294) .

**Comment by XilaiZhang** [comment]
Sorry for the delay, was oncalling for flutter roll and there are roll bin breakages that needed attention, will merge now

**Comment by XilaiZhang** [comment]
yup was about to do that but was in the midst of rolls, will do it now

**Comment by XilaiZhang** [comment]
cp request at https://github.com/flutter/flutter/issues/125591

**Pull request merged**